draft-ietf-jsonpath-base.md

```diff
@@ -289,50 +298,52 @@ JSONPath expressions are applied to a JSON value, the *argument*.
 Within the JSONPath expression, the abstract name `$` is used to refer
 to the *root node* of the argument, i.e., to the argument as a whole.
 
-JSONPath expressions can use the *dot notation*
+JSONPath expressions use the *bracket notation*, for example:
 
 ~~~~
-$.store.book[0].title
+$['store']['book'][0]['title']
 ~~~~
 
-or the more general *bracket notation*
+or the more compact *dot notation*, for example:
 
 ~~~~
-$['store']['book'][0]['title']
+$.store.book[0].title
 ~~~~
 
 to build paths that are input to a JSONPath implementation.
+A single path may use a combination of bracket and dot notations.
+
+A wildcard `*` ({{wildcard}}) may be used to select all children of an
+object or an array.
 
-JSONPath allows the wildcard symbol `*` to select any member of an
-object or any element of an array ({{wildcard}}).
-The descendant operators (which start with `..`) select some or all of the descendants ({{descendant-selectors}}) of a node.
-The array slice
-syntax `[start:end:step]` allows selecting a regular selection of an
-element from an array, giving a start position, an end position, and
+An array slice `start:end:step` ({{slice}}) selects a series of
+elements from an array, giving a start position, an end position, and
 possibly a step value that moves the position from the start to the
-end ({{slice}}).
+end.
 
-Filter expressions are supported via the syntax `?(<boolean expr>)` as in
+Filter expressions `?<boolean expr>` select certain children of an object or array as in
 
 ~~~~
-$.store.book[?(@.price < 10)].title
+$.store.book[?@.price < 10].title
 ~~~~
 
 {{tbl-overview}} provides a quick overview of the JSONPath syntax elements.
```

Comment:
I believe we should be explicit very early about some of the segments being shorthands of the canonical ones.
This allows us to focus on the canonical ones later.

Reply:
I made a start on this in de4e00a42764c9739c2f18ce1559d80c8607f063, but we probably need to do more.

Reply:
I'll keep this conversation open, but we can do that in the next round.

Reply:
@cabo See 48de8fcda69084c2ea850805af8d5041492b1943 for an early statement along the lines you suggested.